File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 8/13/2026, 10:14:45 PM
class Solution {
    public int removeDuplicates(int[] nums) {
        int k=1,i=0,j=1;
        while(j<nums.length)
        {
            if(nums[i]==nums[j])
            j++;
            else
            {
                nums[i+1]=nums[j];
                k++;
                j++;
                i++;
            }
        }
        return k;
        
        
    }
}

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 8/13/2026, 10:14:51 PM
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n=nums.size();
        if(n<=1)
        return n;
        int write=1;
        for(int read=1;read<n;read++)
        {
            if(nums[read]!=nums[read-1])
            {
            nums[write++]=nums[read];
            }
        }
        return write;
        
    }
};